android-instrumentation/src/main/java/co/elastic/apm/android/instrumentation/ui/activities/ActivityLifecyclePlugin.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *	http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package co.elastic.apm.android.instrumentation.ui.activities;
+
+import android.app.Activity;
+
+import androidx.annotation.NonNull;
+
+import net.bytebuddy.build.AndroidDescriptor;
+import net.bytebuddy.description.type.TypeDescription;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import co.elastic.apm.android.instrumentation.ui.common.BaseLifecycleMethodsPlugin;
+
+public class ActivityLifecyclePlugin extends BaseLifecycleMethodsPlugin {
+    private final AndroidDescriptor androidDescriptor;
+
+    public ActivityLifecyclePlugin(AndroidDescriptor androidDescriptor) {
+        this.androidDescriptor = androidDescriptor;
+    }
+
+    @Override
+    public boolean matches(TypeDescription target) {
```

Comment:
Sorry I'm not sure I'm following, I think this might be one of the mismatches between compile time vs runtime instrumentation 😅 - I'm not providing a TypeDescription, Byte Buddy does so for each [plugin](https://javadoc.io/static/net.bytebuddy/byte-buddy/1.12.17/net/bytebuddy/build/Plugin.html) when looping through the classpath looking for classes to instrument, it first calls the `matches` method, and if it returns `true` it proceeds to instrument said type in the `apply` method where we can use ElementMatchers for the advice. Although I might be missing something so please let me know if that's the case.

Reply:
Ah, I got it now. The `matches` method is part of the [`net.bytebuddy.build.Plugin`](https://github.com/raphw/byte-buddy/blob/cf63517990654a6ce85cc24e91c8c129fbd17d35/byte-buddy-dep/src/main/java/net/bytebuddy/build/Plugin.java#L66) interface as it extends the `ElementMatcher` interface.